### ISSUE TITLE
Installation guides: Is `--make-rslave` and `--rbind` really required for mounting pseudo filesystems?

### ISSUE THREAD

**Issue by camoz** [void-linux/void-docs#631]
The installation guides (at least fde.md and chroot.md) use the mount flag `--make-rslave` for mounting the pseudo filesystems `proc`/`sysfs`/`dev`.

E.g. in chroot.md:

> Mount the pseudo-filesystems needed for a chroot:
>
> `# mount --rbind /sys /mnt/sys && mount --make-rslave /mnt/sys`
> `# mount --rbind /dev /mnt/dev && mount --make-rslave /mnt/dev`
> `# mount --rbind /proc /mnt/proc && mount --make-rslave /mnt/proc`

The official live installer however does not include the `--make-rslave` flag:

```
mount_filesystems() {
    for f in sys proc dev; do
        [ ! -d $TARGETDIR/$f ] && mkdir $TARGETDIR/$f
        echo "Mounting $TARGETDIR/$f..." >$LOG
        mount --bind /$f $TARGETDIR/$f >$LOG 2>&1
    done
}
```

Just as a reference, the `arch-chroot` script from Arch does not include `--make-rslave` or  `--make-slave` either.

So my questions:

1. Is `--make-rslave` really necessary or can we drop it from the installation guides?
2. What about `--bind` vs. `--rbind`? For `proc` and `sysfs` there should be no difference, but for `dev` it makes a difference I think. Would it make sense to settle on one variant (either `--rbind dev` from the guides, or `--bind dev` from the installer) to keep things consistent?

Btw I'm not very familiar with how mounts/namespaces work.

**Comment by ahesford** [comment]
Consistency between the manual installation instructions and installer script is not really relevant, because the two branches are used for different purposes in different contexts.

Making these slave mounts is potentially useful during interactive installations because users may wish to change the mount hierarchy by, *e.g.*, mounting efivarfs under `/sys` from outside a chroot while expecting the change inside the chroot. While changing this behavior is not likely to break many workflows, it adds flexibility where it's most needed---to reduce confusion among users who can't appreciate the difference.

This is not important in a scripted installer because the workflow is predefined and known to work as is currently done. If there is an argument for slave mounts in the installer, it should be made in that repo. (I'm not aware of a compelling argument.)

In summary: no, these options aren't really required. Users who understand why they aren't needed and how omitting them change behavior probably don't need to be told. Users who don't understand should probably just cargo cult them.

**Comment by camoz** [comment]
Alright, thanks for the detailed explanation!
